torch/_refs/nn/functional/__init__.py

```diff
@@ -146,6 +152,26 @@ def relu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
     return torch.where(torch.le(a, 0), 0, a)
 
 
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def hardsigmoid(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.hardsigmoid
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    le_neg3 = a <= -3
+    ge_3 = a >= 3
+    zeros = torch.zeros_like(a)
```

Comment:
I think instead of zeros_like and ones_like you should just be able to call where with a scalar (this may require a rebase?). Here's an example: 

https://github.com/pytorch/pytorch/blob/bc1fef96aff4aeffe7a7b39ef8b4ff467860df28/torch/_refs/nn/functional/__init__.py#L312

Reply:
This appears to be a situation where `refs.where` works but `torch.where` does not.

```
    other = torch.true_divide(a, 6) + 0.5
    return refs.where(a <= -3, 0, refs.where(a >= 3, 1, other))
```

Reply:
If that's the case we should just call refs.where explicitly, then. Although I thought scalar support was added to torch.where, too? 

```
torch.where(torch.tensor((True, False)), 0, 1)
: tensor([0, 1])
```